aesthetic/avg_autotags.py

```diff
@@ -23,51 +23,80 @@
+        autotags_90plus = ast.literal_eval(row[5])
+        autotags = row[6].split(',')
         autotags_ctr = collections.Counter()
+        if nghd not in nghd_autotags_urls: nghd_autotags_urls[nghd] = dict()
         for tag in autotags:
             if tag == '':
                 continue
             else:
                 tagname, value = tag.split(':')
-                nghds_autotags[nghd][tagname] += round(float(value), 3)
+                nghds_autotags[nghd][tagname] += round(float(value), 3) 
+                if tagname in nghd_autotags_urls[nghd]: 
+                    nghd_autotags_urls[nghd][tagname] += [url]
+                else: 
+                    nghd_autotags_urls[nghd][tagname] = [url]
+        #'NUM_PHOTOS' is number of photos of that neighborhood            
         nghds_autotags[nghd]['NUM_PHOTOS'] += 1
+        #put in 'NUM_PHOTOS' to nghds_autotags90plus
         nghds_autotags90plus[nghd].update(autotags_90plus + ['NUM_PHOTOS'])
+        #put in 'NUM_PHOTOS', autotags_90plus to overall_autotags_90plus
         overall_autotags_90plus.update(autotags_90plus + ['NUM_PHOTOS'])
- 
-    
+        # this looks like Counter({'NUM_PHOTOS': 15, 'outdoor': 5, 'building':
+        # 2, 'gable': 2, 'nature': 2, 'indoor': 2, 'plant': 2, 'architecture':
+        # 2, 'pink': 1, 'vehicle': 1, 'house': 1, 'pediment': 1, 'frieze': 1,
+        # 'flower': 1, 'green': 1, 'suit': 1, 'banana tree': 1, 'car': 1})
+
     # Compute the overall rates of each tag, so we can subtract them off.
     tempdict = collections.Counter()
     for tag, value in overall_autotags_90plus.items():
         tempdict[tag] = (value*1.0/overall_autotags_90plus['NUM_PHOTOS'])
     del tempdict['NUM_PHOTOS']
     overall_autotags_90plus = tempdict
-
-    output = {}
+    sorted_output = dict()
+    output = dict()
     for nghd, autotags90plus in nghds_autotags90plus.items():
+    	sorted_output[nghd] = dict() #this is output with the max 10 autotags
+        sorted_output[nghd]["autotags_90plus_minusbaseline"] = list()
         autotags = nghds_autotags[nghd]
+        # Counter({'NUM_PHOTOS': 1, 'indoor': 0.925, 'people': 0.895, 'blue':
+        # 0.695, 'sleeping': 0.674})
         num_photos = autotags90plus['NUM_PHOTOS']
-        output[nghd] = {'num_photos': num_photos, 'autotags_90plus': {},
-                'autotags_all': {}, 'autotags_90plus_minusbaseline':{}}
+        sorted_output[nghd]['NUM_PHOTOS'] = num_photos
+        output[nghd] = {'num_photos': num_photos, #'autotags_90plus': {}, 'autotags_all': {}, 
+        'autotags_90plus_minusbaseline':{}}
             # 'autotags_90plus': [], 'autotags_all': []}
         for autotag, count in autotags90plus.most_common():
             if autotag == 'NUM_PHOTOS':
                 continue
             pct_photos = round(count * 1.0 / num_photos, 5)
-            output[nghd]['autotags_90plus'][autotag] = pct_photos
+            # output[nghd]['autotags_90plus'][autotag] = pct_photos
```

Comment:
At Google, we had a rule that was "don't check in commented code." I think that's a pretty good rule. Again, don't be shy about deleting stuff - it's all in git, we can always get it back if we need to.